CONTRIBUTING.md

```diff
@@ -47,6 +47,93 @@ Using commit signoffs tags is mandatory for all commits; we also
 encourage that each commit is small and cohesive. See the next sections for details.
 
 
+### Programming style
+
+#### Code style
+
+Our code style follows [the Google C++ Style
+Guide](https://google.github.io/styleguide/cppguide.html) with a few
+exceptions. To run automatic code style formatting on files you modify, use our
+[clang-format
+template](https://github.com/mendersoftware/mendertesting/blob/master/.clang-format):
+
+```bash
+curl -O https://github.com/mendersoftware/mendertesting/raw/master/.clang-format
+clang-format -i --style=file:.clang-format <MODIFIED_FILES>
+```
+
+Note that the template requires clang-format version 15 or higher.
+
+##### C++ standard
+
+All shared code must use C++ features from no later than the C++11
+standard. Platform code for POSIX platforms is allowed to use C++17 if
+necessary, but C++11 is preferred for consistency with the rest of the code.
+
+#### File names
+
+Files must be named with `.cpp` and `.hpp` extensions. Only use `.h` if the
+header holds C declarations.
+
+##### Indentation
+
+Indentation in Mender code uses 1 tab per level, no spaces. Function arguments
+or initializers that are broken over multiple lines are indented one level.
+
+##### Line length
+
+Our line lengths are capped at 100 characters instead of 80, which is Google's
+cap. When considering line lengths, each tab is considered 4 characters
+wide. [The other guidelines for line
+length](https://google.github.io/styleguide/cppguide.html#Line_Length) still
+apply.
+
+##### Namespaces
+
+All the [rules about
+namespaces](https://google.github.io/styleguide/cppguide.html#Namespaces) apply,
+with one exception: It is allowed to use `using namespace std` to import the
+`std` namespace. All other namespaces must be spelled out where they are used
+(though it is allowed to shorten long names using `namespace short_name =
+long_name`).
+
+##### C headers
+
+In general, C++ libraries and wrappers should be preferred over C
+equivalents. However, it is permitted to include C headers if no good
+alternative is available. This must only be done inside platform code.
```

Comment:
This and also the text above mentions _platform code_. Do we have an explanation of what it means somewhere?

Reply:
It's mentioned under "Organization", but I agree it's not super clear. I'll try to open that paragraph by declaring what shared and platform code sections are.

@TheYoctoJester wanted "Organization" to be below the code style, but this is actually a good reason to put it back on top, so that you read that part first.